test(linter): add test for `disable_for_this_section` fix

Sysix opened the pull request:
tests from https://github.com/oxc-project/oxc/pull/14214

got closed because https://github.com/oxc-project/oxc/pull/14228 got merged first

camc314 commented:
### Merge activity

* **Sep 29, 4:02 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
test(linter): add test for `disable_for_this_section` fix (#14240)

tests from https://github.com/oxc-project/oxc/pull/14214

got closed because https://github.com/oxc-project/oxc/pull/14228 got merged first
